core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -289,7 +289,7 @@ object DumpLogSegments {
           }
           lastOffset = record.offset
 
-          print("offset: " + record.offset + " position: " + validBytes +
+          print("- offset: " + record.offset + " position: " + validBytes +
```

Comment:
I think there's some redundant information that we can exclude if we are printing the batch info: position, timestamp type, magic version, compression codec. Below we can also exclude the producerId, epoch, and the transactional flag.

Also, what do you think about using `|` instead of `-`?